ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts

```diff
@@ -164,7 +166,18 @@ export class SequencingOrderComponent {
     return !this.role.allowedToDoOrderSequencing();
   }
 
-  public canChangeCollectionDate(): boolean{
-      return this.role.allowedToDoOrderSequencing() || this.role.allowToViewSampleLists();
+  canEditCollectionDate(sample : SequencingOrder ) {
+  if (sample.sampleType === 'Normal') {
+  // only a normal sample should have collectionDate editable, for tumor samples it is the date of px
+    if (this.role.canViewSeqOrderStatus()) {
+    // people with this specific view access can only enter a date if there is no previous date entered
+      return (_.isNull(sample.collectionDate) || _.isUndefined(sample.collectionDate) || _.isEmpty(sample.collectionDate));
```

Comment:
I can't remember with the version of underscore we're using, but I thought _.isEmpty returns true for undefined and null as well.  Is this similar to something else in our code?  If so, have a pointer?

Reply:
What's the type of sample.collectionDate?  From the [underscore documentation](https://underscorejs.org/#isEmpty):
isEmpty_.isEmpty(collection) [source](https://underscorejs.org/docs/modules/isEmpty.html)
Returns true if collection has no elements. For **strings** and array-like objects _.isEmpty checks if the length property is 0. For other objects, it returns true if the object has no enumerable own-properties. Note that primitive numbers, booleans and symbols are always empty by this definition.

Reply:
It looks like this is a string.  If that's the case, this is fine as is, but likely still could be reduced to just _.isEmpty(sample.collectionDate) because these test pass:

      expect(_.isEmpty(null)).toBe(true);
      expect(_.isEmpty(undefined)).toBe(true);

Reply:
it is a string, that's why I kept it as is